client/HubApi.ts

```diff
@@ -124,11 +124,16 @@ export default class HubApi<DB extends BehaviorType = BehaviorType.POPUP> { // D
         return this._request(requestBehavior, RequestType.MANAGE_CASHLINK, [request]);
     }
 
-    public checkout<B extends BehaviorType = DB>(
-        request: Promise<CheckoutRequest> | CheckoutRequest,
+    public checkout<R extends CheckoutRequest, B extends BehaviorType = DB>(
+        request: Promise<R> | R,
         requestBehavior: RequestBehavior<B> = this._checkoutDefaultBehavior as any,
-    ): Promise<B extends BehaviorType.REDIRECT ? void : SignedTransaction> {
-        return this._request(requestBehavior, RequestType.CHECKOUT, [request]);
+    ): Promise<B extends BehaviorType.REDIRECT
+        ? void
+        : R extends { version: 2 }
+            ? SimpleResult | SignedTransaction
+            : SignedTransaction // Note: this will not be guaranteed anymore when checkout supports external nim wallets
```

Comment:
Since we still have the Nimiq Network and have direct access to the blockchain to monitor for external tx in the browser, we _do still get_ the signed tx from the network and can simply return it.

So I think this return type will not change.

Reply:
Yes, could be